### PR TITLE
Aegis 377

### PR DESCRIPTION
**Pull request by loahou04**

### CODE REVIEW

**Comment by StommePoes** [comment]
any practical way around nested callbacks? Or would that mean a total rewrite of coachmark?
```
                                   }
                                 }
                               }));
                             }
                           });
                         }
                       });
                     }
                   });
                 }
               });
             }
           });
         }
       });
     }
   });
```

File: README.md [hunk 1]
```diff
@@ -1,106 +1,11 @@
...
-
-Initializes coach mark pointing to the specified element, given an options object,
-with a callback to be executed when the user dismisses the coach mark.  
-
-Special Case: Have a wrapper div as a specified element if the coachmark pointing element is not a block level element,this 
-will allow to adjust css in cases of long texts that needs to be wrapped on two lines and still have proper highlight in smaller screens or mobile views.
-
-```
-
-Options are the following
```
Comment (marcdonovan) [inline README.md]: you can't remove the readme usage stuff unless you put it all in the usage file.

**Comment by loahou04** [comment]
@StommePoes that is just the demo file.  it was just suppose to be used as an example...its not really meant for any type of updates.  we are using the callback function to call the next coachmark in the example.

**Comment by TanujaG** [comment]
approved

File: main.js [hunk 2]
```diff
@@ -1,6 +1,163 @@
...
+				body.offsetHeight,
+				html.clientHeight),
+			rect = this.target.getBoundingClientRect(),
+			// 50 is close enough. This is very browser-specific
+			bottomHalf = rect.bottom - rect.height + 50 + window.pageYOffset > height/2,
+			leftCenterLine = rect.left + rect.width/2 < window.innerWidth/2;
+
+		if (bottomHalf) {
+			placement += 'top';
+		} else {
```
Comment (marcdonovan) [inline main.js]: Jeremy hates these

Comment (StommePoes) [inline main.js]: lawlz. Just hide the else in a ternary operator. Same code, better stink!

Comment (loahou04) [inline main.js]: well Jeremy isn't on this code base :P.

**Pull request merged**